waddrmgr/db.go

```diff
@@ -554,19 +588,24 @@ func deserializeAddressRow(addressID, serializedAddress []byte) (*dbAddressRow,
 // serializeAddressRow returns the serialization of the passed address row.
 func serializeAddressRow(row *dbAddressRow) []byte {
 	// The serialized address format is:
-	//   <addrType><account><addedTime><syncStatus><commentlen><comment>
+	//   <addrType><account><addedTime><syncStatus><watchingOnly><rawdata>
 	//   <rawdata>
 	//
 	// 1 byte addrType + 4 bytes account + 8 bytes addTime + 1 byte
-	// syncStatus + 4 bytes raw data length + raw data
+	// syncStatus + 1 byte watchingOnly + 4 bytes raw data length + raw data
 	rdlen := len(row.rawData)
-	buf := make([]byte, 18+rdlen)
+	buf := make([]byte, 19+rdlen)
 	buf[0] = byte(row.addrType)
 	binary.LittleEndian.PutUint32(buf[1:5], row.account)
 	binary.LittleEndian.PutUint64(buf[5:13], row.addTime)
 	buf[13] = byte(row.syncStatus)
-	binary.LittleEndian.PutUint32(buf[14:18], uint32(rdlen))
-	copy(buf[18:18+rdlen], row.rawData)
+	if row.watchingOnly {
+		buf[14] = trueByte
+	} else {
+		buf[14] = falseByte
+	}
```

Comment:
A small helper func to choose the correct byte here would be a little nicer to read.